debezium-server/debezium-server-core/src/main/java/io/debezium/server/DebeziumServer.java

```diff
@@ -154,7 +154,7 @@ else if (beans.size() > 1) {
     private void configToProperties(Config config, Properties props, String oldPrefix, String newPrefix) {
         for (String name : config.getPropertyNames()) {
             String updatedPropertyName = null;
-            if (!PROPERTY_NAME_PATTERN.asPredicate().test(name)) {
+            if (SHELL_PROPERTY_NAME_PATTERN.asPredicate().test(name)) {
```

Comment:
I don't quite get the change yet. How does it prevent the replacement in "table_name" when part of the key?

Reply:
new Regexp is excluding regular properties which are having `.` character in the name

in term of configuring `snapshot.select.statement.overrides.db_name.table_name` trough shell variable, it is not possible (for the tables having `_` in the name). because we could not know which `_` is meant to be dot `.` which one  is meant to be actual `_` 

for example :
`snapshot_select_statement_overrides_mdb_name_mtable_name` its not possible to correctly replace `_` with `.`

i might be missing  something?

Reply:
> new Regexp is excluding regular properties which are having . character in the name

I must be blind, but where is this exclusion happening? Can you point me to it?

Reply:
line 157 `if (SHELL_PROPERTY_NAME_PATTERN.matcher(name).matches()) {`

[regexp pattern](https://regex101.com/r/tmuNG0/1) `` should only match the properties including [a-zA-Z0-9_] and at-least one `_` character. the regexp pattern is not matching `.`